File: client-clojure/src/sirqul_io_t_platform/specs/connection_group.clj
(ns sirqul-io-t-platform.specs.connection-group
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.account :refer :all]
            [sirqul-io-t-platform.specs.connection :refer :all]
            [sirqul-io-t-platform.specs.group-permissions :refer :all]
            [sirqul-io-t-platform.specs.asset :refer :all]
            [sirqul-io-t-platform.specs.connection-group :refer :all]
            )
  (:import (java.io File)))


(def connection-group-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :owner) account-spec
   (ds/opt :name) string?
   (ds/opt :description) string?
   (ds/opt :connections) (s/coll-of connection-spec)
   (ds/opt :permissions) group-permissions-spec
   (ds/opt :image) asset-spec
   (ds/opt :subGroups) (s/coll-of connection-group-spec)
   })

(def connection-group-spec
  (ds/spec
    {:name ::connection-group
     :spec connection-group-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ConnectionGroup.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ConnectionGroup._

case class ConnectionGroup (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
owner: Option[Account],
name: Option[String],
description: Option[String],
connections: Option[List[Connection]],
permissions: Option[GroupPermissions],
image: Option[Asset],
subGroups: Option[List[ConnectionGroup]])

object ConnectionGroup {
  import DateTimeCodecs._

  implicit val ConnectionGroupCodecJson: CodecJson[ConnectionGroup] = CodecJson.derive[ConnectionGroup]
  implicit val ConnectionGroupDecoder: EntityDecoder[ConnectionGroup] = jsonOf[ConnectionGroup]
  implicit val ConnectionGroupEncoder: EntityEncoder[ConnectionGroup] = jsonEncoderOf[ConnectionGroup]
}
